x509/src/name.rs

```diff
@@ -20,7 +22,73 @@ pub type Name<'a> = RdnSequence<'a>;
 /// ```
 ///
 /// [RFC 5280 Section 4.1.2.4]: https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.4
-pub type RdnSequence<'a> = alloc::vec::Vec<RelativeDistinguishedName<'a>>;
+#[derive(Clone, Debug, Default, PartialEq, Eq, Newtype)]
+pub struct RdnSequence<'a>(alloc::vec::Vec<RelativeDistinguishedName<'a>>);
+
+/// Find the indices of all non-escaped separators.
+fn find(s: &str, b: u8) -> impl '_ + Iterator<Item = usize> {
+    (0..s.len())
+        .filter(move |i| s.as_bytes()[*i] == b)
+        .filter(|i| {
+            let x = i
+                .checked_sub(2)
+                .map(|i| s.as_bytes()[i])
+                .unwrap_or_default();
+
+            let y = i
+                .checked_sub(1)
+                .map(|i| s.as_bytes()[i])
+                .unwrap_or_default();
+
+            y != b'\\' || x == b'\\'
+        })
+}
+
+/// Split a string at all non-escaped separators.
+fn split(s: &str, b: u8) -> impl '_ + Iterator<Item = &'_ str> {
+    let mut prev = 0;
+    find(s, b).chain([s.len()].into_iter()).map(move |i| {
+        let x = &s[prev..i];
+        prev = i + 1;
+        x
+    })
+}
+
+impl<'a> RdnSequence<'a> {
```

Comment:
Unused lifetime:

```suggestion
impl RdnSequence<'_> {
```

...and we should probably also turn on the `unused_lifetimes` lint.

Reply:
I'll turn the lint on in another patch.